assets/javascript/app.js

```diff
@@ -0,0 +1,235 @@
+/**
+ * Function called when clicking the Login/Logout button.
+ */
+// [START buttoncallback]
+function toggleSignIn(event) {
+  event.preventDefault();
+  if (!firebase.auth().currentUser) {
+    // [START createprovider]
+    var provider = new firebase.auth.GoogleAuthProvider();
+    // [END createprovider]
+    // [START addscopes]
+    // provider.addScope('https://www.googleapis.com/auth/contacts.readonly');
+    // [END addscopes]
+    // [START signin]
+    firebase.auth().signInWithPopup(provider).catch(function(error) {
+      // Handle Errors here.
+      var errorCode = error.code;
+      var errorMessage = error.message;
+      // The email of the user's account used.
+      var email = error.email;
+      // The firebase.auth.AuthCredential type that was used.
+      var credential = error.credential;
+      // [START_EXCLUDE]
+      if (errorCode === 'auth/account-exists-with-different-credential') {
+        alert('You have already signed up with a different auth provider for that email.');
```

Comment:
This is fine for development, but when in production showing alert's and prompt's are not seen as professional standards, that is why it is forbidden on project work presentations.